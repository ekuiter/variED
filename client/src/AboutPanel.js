import React from 'react';
import i18n from './i18n';
import {Panel, PanelType} from 'office-ui-fabric-react/lib/Panel';

export default props => (
    <Panel
        isOpen={props.isOpen}
        type={PanelType.smallFixedFar}
        onDismiss={props.onDismiss}
        isLightDismiss={true}
        headerText={i18n.t('aboutPanel.title')}>
        {i18n.t('aboutPanel.content')}
    </Panel>
);