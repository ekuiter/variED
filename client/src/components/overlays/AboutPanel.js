import React from 'react';
import i18n from '../../i18n';
import {Panel, PanelType} from 'office-ui-fabric-react/lib/Panel';
import PropTypes from 'prop-types';

const AboutPanel = props => (
    <Panel
        isOpen={props.isOpen}
        type={PanelType.smallFixedFar}
        onDismissed={props.onDismissed}
        isLightDismiss={true}
        headerText={i18n.t('overlays.aboutPanel.title')}>
        {i18n.t('overlays.aboutPanel.content')}
    </Panel>
);

AboutPanel.propTypes = {
    isOpen: PropTypes.bool.isRequired,
    onDismissed: PropTypes.func.isRequired
};

export default AboutPanel;