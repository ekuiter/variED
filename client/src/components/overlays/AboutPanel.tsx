/**
 * A Fabric panel that contains further information about the application.
 */

import React from 'react';
import i18n from '../../i18n';
import {Panel, PanelType} from 'office-ui-fabric-react/lib/Panel';

export default ({isOpen, onDismissed}: {isOpen: boolean, onDismissed: () => void}) => (
    <Panel
        isOpen={isOpen}
        type={PanelType.smallFixedFar}
        onDismissed={onDismissed}
        isLightDismiss={true}
        headerText={i18n.t('overlays.aboutPanel.title')}>
        {i18n.getElement('overlays.aboutPanel.content')}
    </Panel>
);