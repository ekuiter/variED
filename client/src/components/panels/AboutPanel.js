import React from 'react';
import i18n from '../../i18n';
import {Panel, PanelType} from 'office-ui-fabric-react/lib/Panel';

export default props => (
    <Panel
        isOpen={props.isOpen}
        type={PanelType.smallFixedFar}
        onDismissed={props.onDismissed}
        isLightDismiss={true}
        headerText={i18n.t('panels.aboutPanel.title')}>
        {i18n.t('panels.aboutPanel.content')}
    </Panel>
);