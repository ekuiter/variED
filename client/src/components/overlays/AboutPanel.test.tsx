import React from 'react';
import {shallow} from 'enzyme';
import AboutPanel from './AboutPanel';

describe('AboutPanel', () => {
    it('renders correctly', () => {
        const mock = jest.fn(),
            wrapper = shallow(
                <AboutPanel
                    isOpen={true}
                    onDismissed={mock}/>
            );
        expect(wrapper).toMatchSnapshot();
    });
});