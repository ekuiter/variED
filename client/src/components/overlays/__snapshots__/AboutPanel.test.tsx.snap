// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutPanel renders correctly 1`] = `
<StyledPanelBase
  headerText="About"
  isLightDismiss={true}
  isOpen={true}
  onDismissed={[Function]}
  type={1}
>
  <div>
    <h3>
      variED: The variability editor
    </h3>
    <p>
      View, edit and analyze feature models in the browser - with support for real-time collaboration.
    </p>
    <p>
      This project is a research effort of the 
      <StyledLinkBase
        href="http://www.dbse.ovgu.de/"
        target="_blank"
      >
        DBSE working group
      </StyledLinkBase>
       and has been released under the 
      <StyledLinkBase
        href="https://github.com/ekuiter/variED/blob/master/LICENSE.txt"
        target="_blank"
      >
        LGPL v3 license
      </StyledLinkBase>
      .
    </p>
    <p>
      If you would like to leave any feedback, use our 
      <StyledLinkBase
        href="https://goo.gl/forms/uUJmj68FYir9vEI13"
        target="_blank"
      >
        online form
      </StyledLinkBase>
       or 
      <StyledLinkBase
        href="mailto:kuiter@ovgu.de"
      >
        mail me
      </StyledLinkBase>
      .
    </p>
    <p>
      <StyledLinkBase
        href="https://github.com/ekuiter/variED"
        target="_blank"
      >
        View source code on GitHub
      </StyledLinkBase>
    </p>
  </div>
</StyledPanelBase>
`;
